docs/streams/developer-guide/config-streams.html

```diff
@@ -310,6 +312,40 @@ <h4><a class="toc-backref" href="#id5">bootstrap.servers</a><a class="headerlink
                 <li><a class="reference external" href="../../../javadoc/org/apache/kafka/streams/errors/LogAndFailExceptionHandler.html">LogAndFailExceptionHandler</a>.
                   This handler logs the deserialization exception and then signals the processing pipeline to stop processing more records.</li>
               </ul>
+
+              <p>You can also provide your own customized exception handler besides the library provided ones to meet your needs. For example, you can choose to forward corrupt
+                records into a quarantine topic (think: a “dead letter queue”) for further processing. To do this, use the Producer API to write a corrupted record directly to
+                the quarantine topic. The drawback of this approach is that “manual” writes are side effects that are invisible to the Kafka Streams runtime library,
+                so they do not benefit from the end-to-end processing guarantees of the Streams API:</p>
+
+              <pre class="brush: java;">
+              class SendToDeadLetterQueueExceptionHandler implements DeserializationExceptionHandler {
+                  KafkaProducer&lt;byte[], byte[]&gt; dlqProducer;
+                  String dlqTopic;
+
+                  @Override
+                  public DeserializationHandlerResponse handle(final ProcessorContext context,
+                                                               final ConsumerRecord&lt;byte[], byte[]&gt; record,
+                                                               final Exception exception) {
+
+                      log.warn("Exception caught during Deserialization, sending to the dead queue topic; " +
+                          "taskId: {}, topic: {}, partition: {}, offset: {}",
+                          context.taskId(), record.topic(), record.partition(), record.offset(),
+                          exception);
+
+                      dlqProducer.send(new ProducerRecord&lt;&gt;(dlqTopic, record.timestamp(), record.key(), record.value()));
+
+                      return DeserializationHandlerResponse.CONTINUE;
+                  }
+
+                  @Override
+                  public void configure(final Map&lt;String, ?&gt; configs) {
+                      dlqProducer = .. // get a producer from the configs map
```

Comment:
not sure what the comment means; doesn't the user need to create it's own producer?

Reply:
Where would the producer be closed?

Reply:
The intent is that users can pass in a producer object via the map. But thinking about it twice, I agree with you that `creating a producer with the passed in configuration` sounds better.

Reply:
About when to close a producer: that is a tricky problem, originally I was considering that users need to create the producer before the Streams instance is created, and pass it in to the config map. If we construct the producer in configure, we need to let users remember the constructed producer and close it at task closure.

Reply:
> not sure what the comment means; doesn't the user need to create it's own producer?

@mjsax Thinking it third time ... :P I think passing in a producer object would still be better recommended to users so that users can bookkeep it outside the callee, for its life time management. LMK.

Reply:
Ok. I am fine with this. Do you think the comment is clear enough such that users understand it? Might be good to elaborate a little bit more. Leave it to you to judge.

Reply:
Ack, I will add a few more explanation when merging.